### PR TITLE
Aws session token

### PR DESCRIPTION
**Pull request by EItanya**
# Description

Please include a summary of the changes.

This bug fixes ... \ This new feature can be used to ...

# Context

Users ran into this bug doing ... \ Users needed this feature to ...

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2373

### CODE REVIEW

File: projects/gloo/pkg/plugins/aws/plugin.go [hunk 1]
```diff
@@ -94,14 +91,12 @@ func (p *plugin) ProcessUpstream(params plugins.Params, in *v1.Upstream, out *en
...
-
-	// TODO(ilacakrms): consider if secretRef should be namespace+name
+	var accessKey, sessionToken, secretKey string
 	if upstreamSpec.Aws.SecretRef == nil && !p.enableCredentialsDiscovey {
+
 		return errors.Errorf("no aws secret provided. consider setting enableCredentialsDiscovey to true if you are running in AWS environment")
-	}
-	if upstreamSpec.Aws.SecretRef != nil {
+
+	} else if upstreamSpec.Aws.SecretRef != nil {
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/aws/plugin.go:99]: why this change?

File: projects/gloo/pkg/plugins/aws/plugin_test.go [hunk 2]
```diff
@@ -187,67 +190,58 @@ var _ = Describe("Plugin", func() {
...
 		It("should not process with a function mismatch", func() {
 			destination.DestinationSpec.DestinationType.(*v1.DestinationSpec_Aws).Aws.LogicalName = "somethingelse"
 
-			err := plugin.(plugins.RoutePlugin).ProcessRoute(plugins.RouteParams{VirtualHostParams: vhostParams}, route, outroute)
+			err := awsPlugin.(plugins.RoutePlugin).ProcessRoute(plugins.RouteParams{VirtualHostParams: vhostParams}, route, outroute)
 			Expect(err).To(HaveOccurred())
-			Expect(outroute.TypedPerFilterConfig).NotTo(HaveKey(filterName))
-		})
-
-		It("should not process with no spec", func() {
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/aws/plugin_test.go:217]: why is this removed? i see that's its skipped but not sure what's the context here?

File: projects/gloo/pkg/plugins/aws/plugin_test.go [hunk 3]
```diff
@@ -102,8 +104,9 @@ var _ = Describe("Plugin", func() {
 				},
 				Kind: &v1.Secret_Aws{
 					Aws: &v1.AwsSecret{
-						AccessKey: accessKeyValue,
-						SecretKey: secretKeyValue,
+						AccessKey:    accessKeyValue,
+						SecretKey:    secretKeyValue,
+						SessionToken: sessionTokenValue,
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/aws/plugin_test.go:109]: have one test that proves the session token is optional?

File: projects/gloo/pkg/plugins/aws/plugin.go (hunk 1 above)
Comment (EItanya) [reply projects/gloo/pkg/plugins/aws/plugin.go:99]: The old code was essentially doing the same thing, and this seemed cleaner. Probably not worth the potential bug though, will switch back

File: projects/gloo/pkg/plugins/aws/plugin_test.go (hunk 2 above)
Comment (EItanya) [reply projects/gloo/pkg/plugins/aws/plugin_test.go:217]: It has been skipped since it was copied over from old gloo, so just went ahead and deleted it, happy to undo
